### DISCUSSION TITLE
Enterprise Cloud account to Enterprise Cloud account Collaboration

### DISCUSSION THREAD

**Discussion by aileen-fisher** [community/community#60714] [Enterprise]
WRT two separate agencies (companies if you will) who would like to collaborate on proprietary code.  Could agency1 purchase Enterprise licences for only its own people, and agency2 purchase Enterprise licenses for only its own people, but agency1 could invite members from agency2 to their private repositories?  I'm wondering if this is possible without having to double up on licenses for either agency, and also, it would be easier from a personnel management perspective if each agency could only pay its own bill.  Thank you.

**Comment by Beargruug** [comment]
When it comes to collaborating on proprietary code between two separate agencies, it's important to consider the licensing agreements and terms of the specific software or platform you're using. While I can provide some general information, it's always best to consult with the relevant software providers or legal experts to ensure compliance with licensing requirements.

In the case of GitHub Enterprise licenses, typically each user who needs access to the private repositories would require a license. If Agency 1 and Agency 2 are separate entities with their own set of users, each agency would need to purchase the necessary licenses for their respective users. This means that, in the scenario you described, Agency 1 would purchase Enterprise licenses for its own people, and Agency 2 would purchase Enterprise licenses for its own people.

To enable collaboration between members of Agency 1 and Agency 2, without duplicating licenses, you can utilize GitHub's collaboration features. Agency 1, as the owner of the private repositories, can invite members from Agency 2 to collaborate on those repositories as external collaborators. External collaborators do not require GitHub Enterprise licenses, and they can be granted specific permissions to contribute to the repositories without the need for Agency 2 to purchase additional licenses.

However, it's essential to review the specific terms and conditions of GitHub Enterprise licenses and consult with GitHub or legal experts to ensure compliance with licensing agreements. They can provide accurate information and guidance based on your specific situation and requirements.

**Reply by aileen-fisher** [reply]
Thank you.  Since the code is proprietary (private), and not open-source, does this affect your answer above: 

https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-user-access-to-your-organizations-repositories/adding-outside-collaborators-to-repositories-in-your-organization

"Adding an outside collaborator to a private or internal repository will use one of your paid licenses. For more information, see "[About per-user pricing](https://docs.github.com/en/enterprise-cloud@latest/billing/managing-billing-for-your-github-account/about-per-user-pricing).""
